app/src/main/java/nz/org/cacophony/birdmonitor/Prefs.java

```diff
@@ -774,12 +781,37 @@ public void setDateTimeLastUpdateCheck(long dateTimeLastUpload) {
         setLong(DATE_TIME_LAST_UPDATE_CHECK, dateTimeLastUpload);
     }
 
-    public boolean getFlightModePending() {
-        return getBoolean(FLIGHT_MODE_PENDING_UPDATE);
+    public int getFlightModePending() {
+        return getInt(FLIGHT_MODE_PENDING);
     }
 
-    public void setFlightModePending(boolean pendingUpdate) {
-        setBoolean(FLIGHT_MODE_PENDING_UPDATE, pendingUpdate);
+
+    /*
+    canActivateFlightMode removes the provided flags from preventing flight mode activation
+    and then returns weather or not we can activate flight mode (nothing is requiring connection)
+    */
+    public boolean canActiveFlightMode(int finishedFlags) {
+        int flightPending = getFlightModePending();
+        if ((flightPending & finishedFlags) > 0) {
+            flightPending = setInternetRequired(false, finishedFlags);
+            return flightPending == 0;
+        }
+        return false;
```

Comment:
Could this be replaced with `return setInternetRequired(false, finishedFlags) == 0`?

Reply:
yep will do